src/VisualStudio/Core/Def/Implementation/Experimentation/KeybindingResetDetector.cs

```diff
@@ -319,7 +316,12 @@ public int Exec(ref Guid pguidCmdGroup, uint nCmdID, uint nCmdexecopt, IntPtr pv
             if (pguidCmdGroup == ReSharperCommandGroup && nCmdID >= ResumeId && nCmdID <= ToggleSuspendId)
             {
                 // Don't delay command processing to update resharper status
-                Task.Run(() => InvokeBelowInputPriority(UpdateStateMachine));
+                ThreadingContext.JoinableTaskFactory.RunAsync(async () =>
+                {
+                    await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true, CancellationToken.None);
+
+                    await UpdateStateMachineAsync(CancellationToken.None);
```

Comment:
dumb question.  if UpdateStateMachineAsync needs to be on the main thread... why doesn't it switch to it itself?